lib/charms/mysql/v0/mysql.py

```diff
@@ -1252,6 +1261,70 @@ def offline_mode_and_hidden_instance_exists(self) -> bool:
 
         return matches.group(1) != "0"
 
+    def tls_set_custom(self, ca_path: str, key_path: str, cert_path: str) -> None:
+        """Execute a list of commands to enable TLS.
+
+        Args:
+            ca_path: Path to the CA certificate
+            key_path: Path to the server key_path
+            cert_path: Path to the server certificate
+        """
+        enable_commands = (
+            f"SET PERSIST ssl_ca='{ca_path}';"
+            f"SET PERSIST ssl_key='{key_path}';"
+            f"SET PERSIST ssl_cert='{cert_path}';"
+            "SET PERSIST require_secure_transport=on;"
+            "ALTER INSTANCE RELOAD TLS;"
+        )
+
+        try:
+            self._run_mysqlcli_script(
+                enable_commands,
+                user=self.server_config_user,
+                password=self.server_config_password,
+            )
+        except MySQLClientError:
+            raise MySQLTLSSetCustomConfigError("Failed to set custom TLS configuration")
+
+    def tls_restore_default(self) -> None:
+        """Restore TLS configuration to default."""
+        restore_commands = (
+            "SET PERSIST ssl_ca='ca.pem';"
+            "SET PERSIST ssl_key='server-key.pem';"
+            "SET PERSIST ssl_cert='server-cert.pem';"
+            "SET PERSIST require_secure_transport=off;"
+            "ALTER INSTANCE RELOAD TLS;"
+        )
+
+        try:
+            self._run_mysqlcli_script(
+                restore_commands,
+                user=self.server_config_user,
+                password=self.server_config_password,
+            )
+        except MySQLClientError:
+            raise MySQLTLSRestoreDefaultConfigError
+
+    def kill_unencrypted_sessions(self) -> None:
+        """Kill non local, non system open unencrypted connections."""
+        kill_connections_command = (
+            f"shell.connect('{self.server_config_user}:{self.server_config_password}@{self.instance_address}')",
+            (
+                'processes = session.run_sql("'
+                "SELECT processlist_id FROM performance_schema.threads WHERE "
```

Comment:
[case you're wondered where I've got this from](https://dev.mysql.com/doc/refman/8.0/en/performance-schema-threads-table.html)